lib/shopify_api/rest/request.ex

```diff
@@ -198,4 +243,33 @@ defmodule ShopifyAPI.REST.Request do
       {"#{key}", value}
     end
   end
+
+  @link_regex ~r/<(?<link>.*)>;\s*rel=\"(?<rel>.*)\"/
+
+  @spec extract_next_link(list) :: binary | nil
+  defp extract_next_link(headers) do
+    for {"Link", link_header} <- headers,
+        links <- String.split(link_header, ",") do
+      case Regex.named_captures(@link_regex, links) do
+        %{"link" => link, "rel" => "next"} -> link
+        _ -> nil
+      end
+    end
+    |> Enum.filter(&(not is_nil(&1)))
+    |> List.first()
+  end
+
+  # TODO(BJ) - nail down the exact values if possible
+  @spec extract_body_results({:ok, map()} | any()) :: any()
+  def extract_body_results({:ok, body}), do: body
```

Comment:
Is this function necessary with the changes to `process_response_body` to fully unpack response objects?

Reply:
I think my problem here is that I'm trying to support two types of behaviour: one to keep the tests passing, and one to support the results from my manual testing. The problem test is https://github.com/pixelunion/elixir-shopifyapi/blob/master/test/shopify_api/rest/product_test.exs#L21. I'll try to see if I can refactor the test to behave more like the API.

Reply:
Ok @stewart, I think I've maintained the usefulness of the test while also bringing it more in line with how the API works. Can you please take another look at it and let me know what you think?